lib/serverless/api-gateway.js

```diff
@@ -107,56 +110,82 @@ function normalizeHeaders(event, lowerCaseKey = false) {
  *                    to create a web transaction.
  */
 function isLambdaProxyEvent(event) {
-  return isGatewayV1Event(event) || isGatewayV2Event(event)
+  return isGatewayV1Event(event) || isGatewayV2Event(event) || isAlbEvent(event)
 }
 
+/**
+ * Iterates over the minimum signature properties of an event triggered by API Gateway
+ * to determine the version. V1 and V2 have a lot of overlap, but some signature differences.
+ *
+ * @param {object} targetEvent The event to inspect.
+ * @param {Array} searchFor An array of keys unique to this proxy type
+ * @returns {boolean} Whether this event has matches for the keys we're checking
+ */
+function findProperties(targetEvent, searchFor) {
+  const keys = Object.keys(targetEvent)
+  for (const el of searchFor) {
+    if (keys.indexOf(el) > -1) {
+      return true
+    }
+  }
+  return false
```

Comment:
Isn't this returning true if _any_ of the elements of `searchFor` are present in `targetEvent`? Shouldn't we replicate the functionality of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/isSubsetOf here?

Reply:
It is indeed a test for *any*, which is intentional. With API Gateway v1, properties like `multiValueHeaders` aren't guaranteed to be defined on the event. If we do see that, we can be certain it's a v1 payload, but its absence doesn't mean that a payload isn't v1. If there's a more efficient test pattern than "test for any signature property and bail on a positive," we should use it...but I'm wary of false negatives in this case.

Reply:
Please document these details in the code.